go/tasks/plugins/presto/client/presto_status.go

```diff
@@ -0,0 +1,41 @@
+package client
+
+import (
+	"context"
+	"strings"
+
+	"github.com/lyft/flyteplugins/go/tasks/plugins/svc"
+	"github.com/lyft/flytestdlib/logger"
+)
+
+// This type is meant only to encapsulate the response coming from Presto as a type, it is
+// not meant to be stored locally.
+const (
+	PrestoStatusUnknown   svc.CommandStatus = "UNKNOWN"
+	PrestoStatusQueued    svc.CommandStatus = "QUEUED"
+	PrestoStatusRunning   svc.CommandStatus = "RUNNING"
+	PrestoStatusFinished  svc.CommandStatus = "FINISHED"
+	PrestoStatusFailed    svc.CommandStatus = "FAILED"
+	PrestoStatusCancelled svc.CommandStatus = "CANCELLED"
+)
+
+var PrestoStatuses = map[svc.CommandStatus]struct{}{
+	PrestoStatusUnknown:   {},
+	PrestoStatusQueued:    {},
+	PrestoStatusRunning:   {},
+	PrestoStatusFinished:  {},
+	PrestoStatusFailed:    {},
+	PrestoStatusCancelled: {},
+}
+
+func NewPrestoStatus(ctx context.Context, state string) svc.CommandStatus {
+	upperCased := strings.ToUpper(state)
+	if strings.Contains(upperCased, "FAILED") {
```

Comment:
Not entirely sure, I sort of copied this from the existing qubole_status. Probably not necessary to single this out.

Reply:
Actually, I think it was needed because Hive/Qubole has different error/failure modes and I think this maps them all to a single Failure/Error. Going  to leave this in for now

Reply:
Add this as a comment?